### ISSUE TITLE
NuValidator raising ParsingError, service returning HTML instead of JSON?

### ISSUE THREAD

**Issue by wolfgangrittner** [w3c-validators/w3c_validators#31]
I'm getting a `W3CValidators::ParsingError: unable to parse the response from the validator.` when trying to use the NuValidator like so:
```
validator = ::W3CValidators::NuValidator.new
results = validator.validate_file(file)
```
`file` being a temp file containing HTML5 markup. I also tried validate_text, but got the very same error.

Details of ParsingError raised:
```
 # --- Caused by: ---
     # JSON::ParserError:
     #   757: unexpected token at '<!DOCTYPE html>
     #   <html lang="en"><head><link href="icon.png" rel="icon"><link href="style.css" rel="stylesheet"><title>(X)HTML5 validation results</title></head><body><h1>(X)HTML5 validation results</h1><form method="get" class="simple">....
```

I already checked the request sent to validator.nu, which contains the `out=json` parameter, so should be fine. Any idea what's wrong with the service?

Thanks so much for providing this awesome gem, btw. Looking forward to get this to work properly soon :)

**Comment by doc75** [comment]
Are you sure you are using the latest version of the Gem ?
It looks like an error I fixed in v1.3.1.

Here is what I just tried (assuming you are at the root of w3c_validators git repo):
```ruby
require 'w3c_validators'

file = 'test/fixtures/valid_html5.html'
validator = ::W3CValidators::NuValidator.new
results = validator.validate_file(file)
puts results.inspect
```

Here is the output obtained:
```
#<W3CValidators::Results:0x000000025d7a58 @messages=[], @uri=nil, @checked_by="https://validator.nu/", @doctype=:html5, @css_level=nil, @charset="UTF-8", @validity=true, @debug_messages={}>
```

Let me know if you have other information or if it is not working with 1.3.1

**Comment by wolfgangrittner** [comment]
Hey, thanks for your reply!
Right, didn't mind checking the version. We're forced to an older version of the gem because of the dependency to json 2.x. Another gem in the project I'm trying to use this locks us to json 1.x.
Just out of curiosity, what's the reason for requiring json 2.x? I tried to find out, but failed.

**Comment by doc75** [comment]
Indeed, I recently took over the maintenance of this gem and I put the highest version available at that time.
It seems that it was a bad idea looking at the kind of issue your are facing.
I can relax the constraint by putting ">= 1.8" (I checked and with 1.8.6 all tests are passing).

Would that fit your need or you are on an even older version of json gem ?

**Comment by wolfgangrittner** [comment]
Hey, that would have done the trick, yes. We've a gem that requires `json (~> 1.7)` and are on 1.8.6 currently.

**Comment by doc75** [comment]
Solved with PR #32 
Gem version 1.3.2 published on rubygems.org.
Thanks again for reporting the issue.
